packages/quantic/force-app/main/default/lwc/headlessLoader/headlessLoader.js

```diff
@@ -0,0 +1,150 @@
+import HeadlessPath from '@salesforce/resourceUrl/coveoheadless';
+import AtomicPath from '@salesforce/resourceUrl/atomicutils';
+// @ts-ignore	
+import { loadScript } from 'lightning/platformResourceLoader';
+
+const DEBOUNCE_DELAY = 200;
+let timeout;
+
+/**
+ * Debounces a function execution.
+ * @param {Function} func The function for which to delay execution.
+ * @param {Number} wait The time to delay in milliseconds.
+ */
+const debounce = (func, wait) => {
+  return function executedFunction(...args) {
+    const later = () => {
+      clearTimeout(timeout);
+      func(...args);
+    };
+    clearTimeout(timeout);
+    // eslint-disable-next-line @lwc/lwc/no-async-operation
+    timeout = setTimeout(later, wait);
+  };
+};
+
+/**
+ * Cancels the delayed search query.
+ */
+const cancelInitialSearch = () => {
+  if (timeout) {
+    clearTimeout(timeout);
+    timeout = undefined;
+  }
+}
+
+/**
+ * Dispatches search request.
+ */
+const executeInitialSearch = debounce(() => {
+  window.coveoHeadless.engine.dispatch(
+    CoveoHeadless.SearchActions.executeSearch(
+      CoveoHeadless.AnalyticsActions.logInterfaceLoad()
+    )
+  );
+}, DEBOUNCE_DELAY);
+
+/**
+ * Returns true if registered components are initialized, false otherwise.
+ */
+const areAllComponentsInitialized = () => !window.coveoHeadless.components.find(component => component.initialized === false);
+
+/**
+ * Returns the registered component object if it exists. 
+ * @param element 
+ */
+const getRegisteredComponent = (element) => window.coveoHeadless.components.find((component) => component.element === element);
+
+/**
+ * Loads dependencies and returns an initialized Headless engine. 
+ * @param element 
+ */
+async function initEngine(element) {
+  let engine;
+  try {
+    await loadScript(element, HeadlessPath + '/browser/headless.js');
+    await loadScript(element, AtomicPath + '/atomic-utils.js');
```

Comment:
Could it be more performant to load these scripts when the page loads, rather than waiting for the first call to `initEngine`?

Reply:
I'll look into it 👍